packages/waku/src/lib/vite-rsc/ssr.tsx

```diff
@@ -47,25 +47,45 @@ export async function renderHTML(
 
   // render html
   const bootstrapScriptContent = await loadBootstrapScriptContent();
-  const htmlStream = await renderToReadableStream(<SsrRoot />, {
-    bootstrapScriptContent:
-      getBootstrapPreamble({ rscPath: options?.rscPath || '' }) +
-      bootstrapScriptContent,
-    onError: (e: unknown) => {
-      if (
-        e &&
-        typeof e === 'object' &&
-        'digest' in e &&
-        typeof e.digest === 'string'
-      ) {
-        return e.digest;
-      }
-      console.error('[SSR Error]', captureOwnerStack?.() || '', '\n', e);
-    },
-    ...(options?.nonce ? { nonce: options.nonce } : {}),
-    ...(options?.formState ? { formState: options.formState } : {}),
-  });
-
+  let htmlStream: ReadableStream<Uint8Array>;
+  try {
+    htmlStream = await renderToReadableStream(<SsrRoot />, {
+      bootstrapScriptContent:
+        getBootstrapPreamble({ rscPath: options?.rscPath || '' }) +
+        bootstrapScriptContent,
+      onError: (e: unknown) => {
+        if (
+          e &&
+          typeof e === 'object' &&
+          'digest' in e &&
+          typeof e.digest === 'string'
+        ) {
+          return e.digest;
+        }
+        console.error('[SSR Error]', captureOwnerStack?.() || '', '\n', e);
+      },
+      ...(options?.nonce ? { nonce: options.nonce } : {}),
+      ...(options?.formState ? { formState: options.formState } : {}),
+    });
+  } catch (e) {
+    // SSR empty html and go full CSR on browser, which can revive RSC errors.
+    // TODO: http status
+    const ssrErrorRoot = (
+      <html lang="en">
+        <head>
+          <meta charSet="utf-8" />
+        </head>
+        <body></body>
+      </html>
```

Comment:
Hmm, hard-coding this feel like a hack.
tbh, if we want CSR on error, we should wrap with Suspense, I think.

Reply:
> > how `error.js` convention kicks in on SSR
> 
> Are you sure if it's CSR?

Yes it is. React's error boundary is purely CSR mechanism, so it only kicks in on browser. Here is an example SSR output with `error.js` https://github.com/hi-ogawa/reproductions/tree/main/next-ssr-error